data-protocols/ids/ids-api-multipart-dispatcher-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/sender/MultipartSenderDelegate.java

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright (c) 2022 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.ids.api.multipart.dispatcher.sender;
+
+import de.fraunhofer.iais.eis.DynamicAttributeToken;
+import de.fraunhofer.iais.eis.Message;
+import org.eclipse.dataspaceconnector.ids.api.multipart.dispatcher.sender.response.IdsMultipartParts;
+import org.eclipse.dataspaceconnector.ids.api.multipart.dispatcher.sender.response.MultipartResponse;
+import org.eclipse.dataspaceconnector.spi.types.domain.message.RemoteMessage;
+
+import java.io.InputStream;
+import java.util.List;
+
+/**
+ * Handles message type specific operations when sending an IDS Multipart message.
+ *
+ * @param <M> the type of {@link RemoteMessage} this delegate can handle
+ * @param <R> the expected response type
+ */
+public interface MultipartSenderDelegate<M extends RemoteMessage, R> {
+
+    /**
+     * Builds the IDS multipart header for the request.
+     *
+     * @param request the request.
+     * @param token   the dynamic attribute token.
+     * @return the message header.
+     * @throws Exception if building the message header fails.
+     */
+    Message buildMessageHeader(M request, DynamicAttributeToken token) throws Exception;
+
+    /**
+     * Builds the IDS multipart payload for the request.
+     *
+     * @param request the request.
+     * @return the message payload.
+     * @throws Exception if building the message payload fails.
+     */
+    default String buildMessagePayload(M request) throws Exception {
```

Comment:
I think the signature for now works better, but I noticed that every subclass in fact serialize the body to json, that should be something in charge of the `IdsMultipartSender`, because these delegates should be protocol-agnostic, and the serialization method is definitely protocol-dependent.
However I'm aware that currently is not possible (because the EDR not being an IDS object) and probably not needed as far as we don't support other protocols. But this is something we could take in consideration for the future.

Reply:
well, some of the implementors don't serialize anything (i.e. return `null`), others simply return a non-formatted string (`MultipartContractRejectionSender` returns a reason phrase). 
However I agree that the concrete serialization should be a) harmonized and b) externalized.

Doesn't the `TypeManager` already have the ability to serialize to different serialization contexts?

Reply:
@paullatzelsperger The `TypeManager` can serialize to different contexts. The problem when moving the serialization to the `IdsMultipartSender` is that it would then need to decide which context to use depending on the delegate. This also doesn't seem right to me.